File: src/main/java/com/wooteco/wiki/dto/DocumentFindAllByRecentResponse.java
package com.wooteco.wiki.dto;

import com.wooteco.wiki.entity.Document;
import java.util.List;

public record DocumentFindAllByRecentResponse(List<DocumentRecentResponse> documents) {
    public static DocumentFindAllByRecentResponse of(List<Document> documents) {
        List<DocumentRecentResponse> recentDocuments = documents.stream()
                .map(DocumentRecentResponse::from)
                .toList();
        return new DocumentFindAllByRecentResponse(recentDocuments);
    }
}

File: src/main/java/com/wooteco/wiki/dto/DocumentFindAllByRecentResponse.kt
package com.wooteco.wiki.dto

import com.wooteco.wiki.domain.Document

data class DocumentFindAllByRecentResponse(
    val documents: List<DocumentRecentResponse>
) {
    companion object {
        fun of(documents: List<Document?>): DocumentFindAllByRecentResponse {
            val recentDocuments = documents.filterNotNull().map { DocumentRecentResponse.from(it) }
            return DocumentFindAllByRecentResponse(recentDocuments)
        }
    }
}
